packages/react-databrowser/src/components/databrowser/hooks/use-keys.tsx

```diff
@@ -0,0 +1,83 @@
+import { createContext, useCallback, useContext, useMemo, type PropsWithChildren } from "react"
+import { useDatabrowserStore } from "@/store"
+import { useInfiniteQuery, type UseInfiniteQueryResult } from "@tanstack/react-query"
+
+import { useFetchKeys, type RedisKey } from "./use-fetch-keys"
+
+const KeysContext = createContext<
+  | {
+      keys: RedisKey[]
+      query: UseInfiniteQueryResult
+      refetch: () => void
+    }
+  | undefined
+>(undefined)
+
+export const FETCH_KEYS_QUERY_KEY = "use-fetch-keys"
+
+export const KeysProvider = ({ children }: PropsWithChildren) => {
+  const { search: searchState } = useDatabrowserStore()
+
+  const search = useMemo(
+    () => ({
+      key: searchState.key.includes("*") ? searchState.key : `*${searchState.key}*`,
```

Comment:
if `key` is `user:*`, we're going to also match `*:user:*` here, which it seems is not the expected behaviour?